pkg/action/diff.go

```diff
@@ -61,39 +61,87 @@ func Diff(ctx context.Context, c Config) (*bincapz.Report, error) {
 		Modified: map[string]*bincapz.FileReport{},
 	}
 
+	processSrc(ctx, c, src, dest, d)
+	processDest(ctx, c, src, dest, d)
+	inferMoves(ctx, c, d)
+
+	return &bincapz.Report{Diff: d}, err
+}
+
+func processSrc(
+	ctx context.Context,
+	c Config,
+	src, dest map[string]*bincapz.FileReport,
+	d *bincapz.DiffReport,
+) {
 	// things that appear in the source
-	for relPath, fr := range from {
-		tr, exists := to[relPath]
+	for relPath, fr := range src {
+		tr, exists := dest[relPath]
 		if !exists {
 			d.Removed[relPath] = fr
 			continue
 		}
-		// We've now established that file exists in both source & destination
-		if fr.RiskScore < c.MinFileScore && tr.RiskScore < c.MinFileScore {
-			clog.FromContext(ctx).Info("diff does not meet min trigger level", slog.Any("path", tr.Path))
-			continue
-		}
+		handleFile(ctx, c, fr, tr, relPath, d)
+	}
+}
 
-		rbs := &bincapz.FileReport{
-			Path:              tr.Path,
-			Behaviors:         map[string]*bincapz.Behavior{},
-			PreviousRiskScore: fr.RiskScore,
-			PreviousRiskLevel: fr.RiskLevel,
-			RiskLevel:         tr.RiskLevel,
-			RiskScore:         tr.RiskScore,
-		}
+func handleFile(
+	ctx context.Context,
+	c Config,
+	fr, tr *bincapz.FileReport,
+	relPath string,
+	d *bincapz.DiffReport,
+) {
+	// We've now established that file exists in both source & destination
+	if fr.RiskScore < c.MinFileScore && tr.RiskScore < c.MinFileScore {
+		clog.FromContext(ctx).Info("diff does not meet min trigger level", slog.Any("path", tr.Path))
+		return
+	}
 
-		// if source behavior is not in the destination
-		for key, b := range fr.Behaviors {
-			if _, exists := tr.Behaviors[key]; !exists {
-				b.DiffRemoved = true
-				rbs.Behaviors[key] = b
-			}
+	rbs := createFileReport(tr, fr)
+
+	// if source behavior is not in the destination
+	for _, fb := range fr.Behaviors {
+		if !behaviorExists(fb, tr.Behaviors) {
+			fb.DiffRemoved = true
+			rbs.Behaviors = append(rbs.Behaviors, fb)
 		}
+	}
+
+	d.Modified[relPath] = rbs
+}
 
-		d.Modified[relPath] = rbs
+func createFileReport(
+	tr, fr *bincapz.FileReport,
+) *bincapz.FileReport {
+	return &bincapz.FileReport{
+		Path:              tr.Path,
+		Behaviors:         []*bincapz.Behavior{},
+		PreviousRiskScore: fr.RiskScore,
+		PreviousRiskLevel: fr.RiskLevel,
+		RiskLevel:         tr.RiskLevel,
+		RiskScore:         tr.RiskScore,
 	}
+}
 
+func behaviorExists(
```

Comment:
For consistency, please keep function definitions to a single line.

This also encourages smaller variable names and encourages people to not introduce functions with an excessive number (>4) of positional arguments. We should add a lint check for that :)

Reply:
Makes sense -- some of these new functions have >4 arguments but going forward I'll keep that in mind.